clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatMonitoringService.java

```diff
@@ -455,12 +457,14 @@ private class HeartbeatLagLoggingThread extends Thread {
     @Override
     public void run() {
       while (!Thread.interrupted()) {
-        checkAndMaybeLogHeartbeatDelay();
         try {
+          checkAndMaybeLogHeartbeatDelay();
           TimeUnit.SECONDS.sleep(DEFAULT_LAG_LOGGING_THREAD_SLEEP_INTERVAL_SECONDS);
         } catch (InterruptedException e) {
           // We've received an interrupt which is to be expected, so we'll just leave the loop and log
           break;
+        } catch (Exception e) {
```

Comment:
Do we need to emit metrics to track this HB service as it is critical? 
If it keeps throwing exception in this function, it can result in a big issue in HB lag tracking.

Reply:
So the symptom would be increasing lag, we can check logs for the following message would help in debugging in those cases?

Reply:
It's a good question and I wonder if we should do some tweaking in how this works.  Right now if the alert goes blank it can mean that the store was either deleted or the reporting is broken.

Right now the alert is structured as anything ><big number> triggers an alert.  I had heretofore resisted the idea of putting a negative number on this similar to our ingestion stats error codes as our alerting system didn't really handle this sort of thing well, but a lot has changed lately.  So maybe theres opportunity to publish into this stat some kind of negative number error code that will emit even if the thread dies?

Reply:
You mean to say emit negative number into lag when the thread dies (case of Throwable), thats reasonable ask. But I think Gaojie asking about catching other types of exceptions which could be temporary in nature, do we need a stat for that or not.

Reply:
Yeah, I was thinking two layers of monitoring:
1. HB service itself.
2. HB lags.

If HB service is not running, do we have a way to emit a negative lag?

For #1, I was thinking some metrics to track the healthiness of the `HeartbeatMonitoringService` in a similar way as [`ParticipantStoreConsumptionTask` ](https://github.com/linkedin/venice/blob/main/clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ParticipantStoreConsumptionTask.java#L64)